browser/search_engines/android/java/src/org/chromium/brave/browser/search_engines/settings/CustomSearchEnginesUtil.java

```diff
@@ -0,0 +1,173 @@
+            Log.e(TAG, "Error saving search engines", e);
+        }
+    }
+
+    public static void addCustomSearchEngine(String searchEngineKeyword) {
+        List<String> customSearchEnginesList = getCustomSearchEngines();
+        if (customSearchEnginesList.isEmpty()) {
+            customSearchEnginesList = new LinkedList<>();
+        }
+        if (!customSearchEnginesList.contains(searchEngineKeyword)) {
+            customSearchEnginesList.add(searchEngineKeyword);
+            saveCustomSearchEngines(customSearchEnginesList);
+        }
+    }
+
+    public static List<String> getCustomSearchEngines() {
+        List<String> customSearchEnginesList = new LinkedList();
+        String savedSearchEngines =
+                ChromeSharedPreferences.getInstance().readString(CUSTOM_SEARCH_ENGINES, null);
+
+        if (savedSearchEngines == null) {
+            return customSearchEnginesList;
+        }
+
+        try {
+            // Wrap the array in an object to make it valid JSON
+            JSONObject wrapper =
+                    new JSONObject("{\"customSearchEngines\":" + savedSearchEngines + "}");
+            JSONArray searchEnginesArray = wrapper.getJSONArray("customSearchEngines");
+
+            for (int i = 0; i < searchEnginesArray.length(); i++) {
+                customSearchEnginesList.add(searchEnginesArray.getString(i));
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return customSearchEnginesList;
+    }
+
+    public static boolean isCustomSearchEngineAdded(String searchEngineKeyword) {
+        List<String> customSearchEnginesList = getCustomSearchEngines();
+        return !customSearchEnginesList.isEmpty()
+                && customSearchEnginesList.contains(searchEngineKeyword);
+    }
+
+    public static void removeCustomSearchEngine(String searchEngineKeyword) {
+        List<String> customSearchEnginesList = getCustomSearchEngines();
+        if (!customSearchEnginesList.isEmpty()
+                && customSearchEnginesList.contains(searchEngineKeyword)) {
+            customSearchEnginesList.remove(searchEngineKeyword);
+            saveCustomSearchEngines(customSearchEnginesList);
+        }
+    }
+
+    public static void updateCustomSearchEngine(
+            String searchEngineKeyword, String newSearchEngineKeyword) {
+        List<String> customSearchEnginesList = getCustomSearchEngines();
+        if (!customSearchEnginesList.isEmpty()
+                && customSearchEnginesList.contains(searchEngineKeyword)) {
+            int index = customSearchEnginesList.indexOf(searchEngineKeyword);
+            if (index != -1) {
+                customSearchEnginesList.set(index, newSearchEngineKeyword);
+            }
+            saveCustomSearchEngines(customSearchEnginesList);
+        }
+    }
+
+    public static boolean isSearchQuery(String text) {
+        return text.contains("%s");
+    }
+
+    public static boolean isValidUrl(String url) {
```

Comment:
where is this method called, why are we dealing with a string instead of a GURL and what exactly is this validating?

Reply:
i have updated the check here : https://github.com/brave/brave-core/blob/7706564bdb5bc6f33bb885176dd71c072e4b8c58/browser/search_engines/android/java/src/org/chromium/brave/browser/search_engines/settings/AddCustomSearchEnginePreferenceFragment.java#L310